### PR TITLE
🐛 Source OpenWeather: Update openweather onecall api to 3.0

### PR DESCRIPTION
**Pull request by pierre-borckmans**
## What
The openweather onecall api only works with the 3.0 version of the api, unless I'm missing something as explained here:
https://openweathermap.org/price

## How
This PR simply changes the base openweather URL to use the 3.0 api

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@pierre-borckmans please update the Dockerfile version and the doc too

**Comment by pierre-borckmans** [comment]
@marcosmarxm Done 👍

**Comment by pierre-borckmans** [comment]
@marcosmarxm More details can be found here, which seems to confirm my observations about OpenWeahter deprecating the 2.5 API for the `OneCall` endpoint:

https://community.home-assistant.io/t/cannot-setup-openweathermap/450817

TL;DR: When creating new API keys (which I did when testing this), only the `3.0` API is supported.

**Comment by pierre-borckmans** [comment]
@marcosmarxm Are those test failure related to my changes? I wouldn't expect so, but let me know :)

**Comment by pierre-borckmans** [comment]
@marcosmarxm I see one of the tests fails because of ```***"type": "TRACE", "trace": ***"type": "ERROR", "emitted_at": 1661967159125.733, "error": ***"message": "Invalid API key. Please see http://openweathermap.org/faq#error401 for more info.", "internal_message": "401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall?appid=****&lat=45.7733258&lon=4.8398336&units=standard"```

Possibly the app id that is used for this test is an old one.
When creating a new API key (with a newly registered user), it will work with the `3.0` API, and not with the `2.5`.

What do you think is the best way to move forward?

**Review by marcosmarxm — APPROVED** [review]
